### PR TITLE
Update CredentialRequest schema

### PR DESCRIPTION
**Pull request by iagormoraes**
[Ticket](https://trello.com/c/I2MMEwrB/5601-update-docs-to-show-updated-credentialrequest-object-with-description-field)

## Summary
Updates the CredentialRequest type example on section of API.

Screenshot 2023-07-10 at 10 51 38

image

image

## Changes
Updates the API and terminology pages.

## Testing
Tested locally.

## Checklist

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas _to a borderline excessive amount_
- [x] I have made any relevant corresponding changes to the documentation including the project readme.
- [x] I have run and tested the changes locally
- [ ] If it is a core feature, I have added an appropriate amount of unit tests.
- [ ] Any dependent changes have been merged and published in upstream projects

### CODE REVIEW

File: docs/api-overview.mdx [hunk 1]
```diff
@@ -398,18 +399,17 @@ This endpoint returns a list of credential requests that the partner has defined
...
-    "type": string, // credential type
-    "issuers": string[], // list of acceptable brandIds; if empty, all issuer brands are valid
-    "required": boolean, // whether the credential is required
-  }[]
+  "credentialRequests": CredentialRequest[]
 }
 ```
 
+Type reference for:
+- [CredentialRequest](https://docs.verified.inc/types/interfaces/credentialrequest.html)
```
Comment (jacobleesinger) [inline docs/api-overview.mdx:409]: I think this is generated from the old types repo that we used in our old architecture. I'm not sure if we have anything like it for the current architecture

**Review by jacobleesinger — CHANGES_REQUESTED** [review]

File: docs/api-overview.mdx [hunk 2]
```diff
@@ -78,7 +78,8 @@ A credential requests encodes which <Tip type="credential">credentials</Tip> you
 {
   type: string, // the type of credential data being requested
   issuers: string[], // list of acceptable brandIds; if empty, all issuer brands are valid
-  required?: boolean // if credential is required (default is true)
+  required?: boolean, // if credential is required (default is true)
+  description?: string // the given description to be explained
```
[the remark this answers is not in the document]
Comment (iagormoraes) [reply docs/api-overview.mdx:82]: Looks much more understandable now, thanks!

File: docs/api-overview.mdx (hunk 1 above)
Comment (iagormoraes) [reply docs/api-overview.mdx:409]: Should then I add the same block piece as in Check User Credentials? Didn't saw a api references either.

Comment (jacobleesinger) [reply docs/api-overview.mdx:409]: Yeah that's fine

**Comment by jacobleesinger** [comment]
@rsmets why is this a "safe change"?

File: docs/api-overview.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (jacobleesinger) [reply docs/api-overview.mdx]: Did you make the change? I don't see it in the code

**Comment by rsmets** [comment]
> @rsmets why is this a "safe change"?

https://docs.gitstream.cm/automations/approve-safe-changes/

I was on the fence about enabling this one. I'll just remove it.

File: docs/terminology.mdx [hunk 3]
```diff
@@ -96,7 +96,8 @@ The full details of the request object aren't that important or helpful to know,
     {
       "type": "SsnCredential",
       "issuers": ["d14ca24c-5323-4b79-9dfa-6471b014d68c"],
-      "required": true
+      "required": true,
+      "description": "Most services that ask for SSN explain why they're asking for it, since people are nervous about providing it."
```
Comment (rsmets) [inline docs/terminology.mdx:100]: I think this should be an example description to better match the rest of the example json.

```suggestion
      "description": "We use your social security number for identity verification and tax reporting."
```

File: docs/reusables/credential-request-code-snippet.mdx [hunk 4]
```diff
@@ -0,0 +1,8 @@
+```typescript title="CredentialRequest"
+{
+  type: string, // the type of credential data being requested
+  issuers: string[], // list of acceptable brandIds; if empty, all issuer brands are valid
+  required?: boolean, // if credential is required (default is true)
```
Comment (rsmets) [inline docs/reusables/credential-request-code-snippet.mdx:5]: I actually do not think it defaults to true anymore. Let's just drop that bit. 

```suggestion
  required?: boolean, // if the credential is required
```

File: docs/reusables/credential-request-code-snippet.mdx [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+  description?: string // the given description to be explained
```
Comment (rsmets) [inline docs/reusables/credential-request-code-snippet.mdx:6]:
```suggestion
  description?: string // a description field that is shown to the user; can be used to explain why the information is being requested
```

**Review by rsmets — CHANGES_REQUESTED** [review]

**Comment by jacobleesinger** [comment]
> > @rsmets why is this a "safe change"?
> 
> @jacobleesinger https://docs.gitstream.cm/automations/approve-safe-changes/
> 
> I was on the fence about enabling this one. I'll just remove it.

That might be fine for documentation-only changes in most repos (updating comments, a readme, etc) but I think public docs should have a human reviewer unless it's like a 1-line typo fix or something

**Comment by rsmets** [comment]
> > > @rsmets why is this a "safe change"?
> > 
> > 
> > @jacobleesinger https://docs.gitstream.cm/automations/approve-safe-changes/
> > I was on the fence about enabling this one. I'll just remove it.
> 
> That might be fine for documentation-only changes in most repos (updating comments, a readme, etc) but I think public docs should have a human reviewer unless it's like a 1-line typo fix or something

Yeah. Just opting to remove that docs and tests rule. https://github.com/UnumID/cm/pull/4 @jacobleesinger

**Review by rsmets — APPROVED** [review]

**Comment by rsmets** [comment]
Feel free to merge if Jacob's requested changes have been done (pr is saying they are not resolved yet)

File: docs/api-overview.mdx (hunk 2 above)
[the remark this answers is not in the document]
Comment (iagormoraes) [reply docs/api-overview.mdx]: @jacobleesinger I reused the component on `credential-request-code-snippet.mdx`, see if make sense to reuse that.

**Review by jacobleesinger — APPROVED** [review]

**Pull request merged**